android/src/firebase/config/TitaniumFirebaseConfigModule.java

```diff
@@ -4,148 +4,176 @@
+                    result.put(PROPERTY_RESULT, task.getResult());
+                } else {
+                    result.put(PROPERTY_SUCCESS, false);
+                    if (task.getException() != null) {
+                        result.put(PROPERTY_ERROR, task.getException().getMessage());
+                    }
+                }
+                callback.callAsync(getKrollObject(), result);
+            }
+        });
+    }
+
+    @Kroll.method
+    public void setMinimumFetchIntervalInSeconds(int minimumFetchInterval) {
+        FirebaseRemoteConfigSettings configSettings = new FirebaseRemoteConfigSettings.Builder()
+                .setMinimumFetchIntervalInSeconds(minimumFetchInterval)
+                .build();
+        FirebaseRemoteConfig.getInstance().setConfigSettingsAsync(configSettings);
+    }
+
+    @Kroll.method
+    public void setDefaults(Object params) throws Exception {
+        if (params instanceof String) {
+            FirebaseRemoteConfig.getInstance().setDefaultsAsync(TiRHelper.getResource("xml." + TiConvert.toString(params)));
+        } else if (params instanceof HashMap) {
+            FirebaseRemoteConfig.getInstance().setDefaultsAsync((HashMap) params);
+        } else {
+            throw new Exception("Invalid defaults provided. Please either pass a dictionary or string");
+        }
+    }
+
+    @Kroll.method
+    public void activateFetched(KrollFunction callback) {
+        FirebaseRemoteConfig.getInstance().activate().addOnCompleteListener(task -> {
+            if (callback != null) {
+                KrollDict result = new KrollDict();
+                if (task.isSuccessful()) {
+                    result.put(PROPERTY_SUCCESS, true);
+                    result.put(PROPERTY_RESULT, task.getResult());
+                } else {
+                    result.put(PROPERTY_SUCCESS, false);
+                    if (task.getException() != null) {
+                        result.put(PROPERTY_ERROR, task.getException().getMessage());
+                    }
+                }
+                callback.callAsync(getKrollObject(), result);
+            }
+        });
+    }
+
+    @Kroll.method
+    public KrollDict configValueForKey(String key) {
+        String stringValue = FirebaseRemoteConfig.getInstance().getString(key);
+        boolean boolValue = FirebaseRemoteConfig.getInstance().getBoolean(key);
+        double numberValue = FirebaseRemoteConfig.getInstance().getDouble(key);
+
+        KrollDict result = new KrollDict();
+        result.put("bool", boolValue);
+        result.put("number", numberValue);
+
+        if (stringValue != null) {
+            result.put("string", stringValue);
+        }
+
+        return result;
+    }
+
+    @Kroll.setProperty
+    public void enableRealtimeUpdates(Boolean value) {
+        if (value) {
+            configUpdateListener = FirebaseRemoteConfig.getInstance().addOnConfigUpdateListener(new ConfigUpdateListener() {
```

Comment:
On iOS, I've added a check to prevent duplicate `enableRealtimeUpdates` calls from adding multiple listeners, as this can cause duplicate events sometimes. Not sure if Firebase keeps track internally, but just in case, the following check could help (before the if-check):

```java
if (configUpdateListener != null) {
    configUpdateListener.remove();
}
```

Reply:
good idea, will add that in a bit

Reply:
that didn't work on Android. I've added 4 listeners in JS and it still fired it 4 times. I use the other way now
```
if (configUpdateListener == null) {
...
```

and that works. It will only add one listener even if I add 4 in JS